android/java/org/chromium/chrome/browser/vpn/wireguard/WireguardConfigUtils.java

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.vpn.wireguard;
+
+import static org.chromium.chrome.browser.vpn.wireguard.WireguardUtils.getInterface;
+import static org.chromium.chrome.browser.vpn.wireguard.WireguardUtils.getPeers;
+
+import android.content.Context;
+
+import com.wireguard.config.BadConfigException;
+import com.wireguard.config.Config;
+
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.vpn.wireguard.TunnelModel;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class WireguardConfigUtils {
+    public static Config createConfig(Context context, String address, String host,
+            String clientPrivateKey, String serverPublicKey)
+            throws IOException, BadConfigException {
+        File file = fileForConfig(context);
+        if (!file.createNewFile()) {
+            throw new IOException("Configuration file already exists");
+        }
+        FileOutputStream fileOutputStream = new FileOutputStream(file, false);
+        Config config = new Config.Builder()
+                                .setInterface(getInterface(address, clientPrivateKey))
+                                .addPeers(getPeers(host, serverPublicKey))
+                                .build();
+        fileOutputStream.write(config.toWgQuickString().getBytes(StandardCharsets.UTF_8));
+        return config;
+    }
+
+    public static void deleteConfig(Context context) throws Exception {
+        File file = fileForConfig(context);
+        if (!file.delete()) {
+            throw new IOException("Cannot delete configuration file");
+        }
+    }
+
+    public static Config loadConfig(Context context) throws Exception {
+        FileInputStream fileInputStream = new FileInputStream(fileForConfig(context));
+        return Config.parse(fileInputStream);
+    }
+
+    public static boolean isConfigExist(Context context) {
+        return fileForConfig(context).exists();
+    }
+
+    private static File fileForConfig(Context context) {
```

Comment:
This should either use `getNoBackupFilesDir` or `getDataDir` since `getFilesDir` is not guaranteed to be a private directory. Ideally, this key should stored encrypted with an EncryuptedFile (see https://developer.android.com/guide/topics/security/cryptography#write-file or https://developer.android.com/guide/topics/security/cryptography#read-file on how to use these primitives )

Reply:
@thypon we can't use getDataDir based on https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/content/Context.java#1132

Reply:
`getDataDir` may also store the file on adoptable storage (eg. external SD card)

@diracdeltas is it acceptable to store VPN credentials on adoptable storage?

@deeppandya I feel if we want to store on adoptable storage we should at least encrypt with the EncryptedFile mechanism.

Reply:
Updated code with `getNoBackupFilesDir`

Reply:
@thypon created a Follow up issue to add encrypted file for wireguard config file : https://github.com/brave/brave-browser/issues/22383